### DISCUSSION TITLE
Python-udf stuck while loading pretrained sklearn models via pickle

### DISCUSSION THREAD

**Discussion by blickensdoerferLuk** [davidfrantz/force#324] [Q&A]
Hi David, Hi Everyone,

I am trying to apply a sklean model inside my Python-UDF to each pixel. I am therefore trying to load a trained sklean model via pickle. 
So in my header (before forcepy_init and forcepy_pixel) I try to load the model via:

```python

# load trained RF model
pickel_file = open(model_file_path, 'rb')
clf = pickle.load(pickel_file)
pickel_file.close()
```

While the file is being opened, FORCE (or rather python) gets stuck at pickle.load() with no error and without exiting….

Any experience or ideas why this might be the case?

I am using the davidfrantz/force:latest docker image (but additionally installed scikit-learn to be available in to python).

Any ideas or hints are much appreciated!!
Thanks,
Lukas

****
Update:
****

I have done some trial and error testing, and pickle loads other objects fine (i.e. numpy array or even a not fitted sklearn classifier). But as soon as the classifier is fitted, pickle gets stuck. I thought size might be an issue, but it is only 7.5MB.

I also started the docker in interactive mode and ran python. Also here, the object is loaded without any problem.

**Comment by davidfrantz** [comment]
Hi Lukas,

I am afraid, I cannot help much regarding this point. But I remember that some errors resulted in the python interpreter to hang (like you describe it). I thought we added some trycatch routines to at least exit gracefully with an error message.

Probably @janzandr can help here?

Is there a possibility, that there might be some sort of version conflict between the package (I assume installed in your user space) and the stuff that is installed in the container (python version as such, packages like numpy)?

Best, David

**Reply by janzandr** [reply]
Hi, I'm also not sure, what could cause this. You could try to **"Save Your Model with joblib"**, instead of **pickle**:
https://machinelearningmastery.com/save-load-machine-learning-models-python-scikit-learn/
